### PR TITLE
[3] Enhance out of stock message

### PR DESCRIPTION
**Pull request by JimmyHoenderdaal**
makes the out of stock message in the cart look nicer
image

### CODE REVIEW

**Comment by Roene-JustBetter** [comment]
Can you add some info and screenshots?

I think we also want this in V2?

**Review by Roene-JustBetter — APPROVED** [review]

File: resources/views/cart/partials/product/description.blade.php [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+        <div>
+            <a :href="item.product.url_key + item.product.url_suffix | url" dusk="cart-item-name">@{{ item.product.name }}</a>
             <div v-for="option in item.configurable_options">
                 @{{ option.option_label }}: @{{ option.value_label }}
             </div>
             <div v-for="option in item.customizable_options">
                 @{{ option.label }}: @{{ option.values[0].label || option.values[0].value }}
             </div>
-            <div class="text-red-600" v-if="!canOrderCartItem(item)">
+            <div v-if="!canOrderCartItem(item)" class="text-ct-error text-balance bg-ct-error/5 my-2 max-w-sm p-2">
```
Comment (BobWez98) [inline resources/views/cart/partials/product/description.blade.php:12]:
```suggestion
            <div v-if="!item.is_available" class="text-ct-error text-balance bg-ct-error/5 my-2 max-w-sm p-2">
```

File: resources/views/cart/partials/product/image.blade.php [hunk 2]
```diff
@@ -1,5 +1,10 @@
 <td class="h-24 w-32 !pl-0 max-md:flex max-md:w-1/2">
-    <img v-if="item.product.image" class="object-contain" :alt="item.product.name" :src="'/storage/{{ config('rapidez.store') }}/resizes/200/magento' + item.product.image.url.replace(config.media_url, '') + '.webp'">
+    <img
+        v-if="item.product.image"
+        v-bind:class="{'opacity-20' : !canOrderCartItem(item)}"
```
Comment (BobWez98) [inline resources/views/cart/partials/product/image.blade.php:4]:
```suggestion
        v-bind:class="{'opacity-20' : !item.is_available}"
```

**Review by BobWez98 — CHANGES_REQUESTED** [review]
Reference; https://github.com/rapidez/core/pull/627/files#diff-9da9c7eb5d8ffc35371000ac0ceb2c0f5b43725a9dfc4312be677e84a22cf3dbR18

**Review by Roene-JustBetter — DISMISSED** [review]

**Review by jordythevulder — DISMISSED** [review]

**Review by BobWez98 — DISMISSED** [review]
Let's not merge this untill https://github.com/rapidez/core/pull/627 and https://github.com/rapidez/core/pull/628

**Comment by royduin** [comment]
Conflicts 😇 and the core requirement should be bumped: https://github.com/rapidez/core/releases/tag/2.17.0

**Review by Roene-JustBetter — APPROVED** [review]

File: resources/views/cart/partials/product/remove-button.blade.php [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+    :notify="{ message: item.product.name + ' ' + config.translations.cart.remove }"
     :callback="updateCart"
     :error-callback="checkResponseForExpiredCart"
-    v-slot="{ mutate }"
 >
-    <button v-on:click="mutate" class="text-ct-inactive mt-1 text-xs hover:underline" :dusk="'item-delete-' + index">
+    <button
+        v-on:click="mutate"
+        class="text-ct-inactive mt-1 text-xs hover:underline"
+        v-bind:class="{'!text-ct-error !underline' : !canOrderCartItem(item)}"
```
Comment (BobWez98) [inline resources/views/cart/partials/product/remove-button.blade.php:12]: I think this should be `!item.is_available` now.

**Review by BobWez98 — CHANGES_REQUESTED** [review]

**Comment by BobWez98** [comment]
Checks are failing.

File: resources/views/cart/partials/product/price.blade.php [hunk 4]
```diff
@@ -7,7 +7,9 @@
     </div>
 </td>
 <td class="flex items-center font-medium max-md:w-1/3 md:table-cell *:mx-auto">
-    <x-rapidez-ct::input.quantity/>
+    <template v-if="item.is_available">
+        <x-rapidez-ct::input.quantity/>
```
Comment (Roene-JustBetter) [inline resources/views/cart/partials/product/price.blade.php:11]: Out of scope but we can use the quantity select from the rapidez core?

**Review by Roene-JustBetter — DISMISSED** [review]

File: resources/views/cart/partials/product/price.blade.php (hunk 4 above)
Comment (JimmyHoenderdaal) [reply resources/views/cart/partials/product/price.blade.php:11]: Just like the other feedback, its out of scope and we have issues ready for this subject

**Review by BobWez98 — DISMISSED** [review]

File: lang/nl.json [hunk 5]
```diff
@@ -64,6 +61,8 @@
     "Thank you for subscribing": "Bedankt voor uw inschrijving",
     "Thank you for your order": "Bedankt voor uw bestelling",
     "This is a business account": "Dit is een zakelijke account",
+    "This product is out of stock": "Dit product is op dit moment niet op voorraad",
+    "This product it out of stock, remove it to continue your order.": "Dit product is niet op voorraad, verwijder het om verder te gaan met je bestelling.",
```
Comment (jordythevulder) [inline lang/nl.json:65]: Uw / Je translations

Comment (royduin) [reply lang/nl.json]: "u" is to formal here, let's stick to "je" for everything.

**Comment by royduin** [comment]
So 2.x doesn't need this (anymore)?

**Pull request merged**